src/cargo/util/auth/mod.rs

```diff
@@ -236,6 +237,17 @@ pub fn registry_credential_config_raw(
         return Ok(cfg.clone());
     }
     let cfg = registry_credential_config_raw_uncached(gctx, sid)?;
+    if let Some(RegistryConfig {
+        token: Some(token), ..
+    }) = &cfg
+    {
+        check_token(&token.val.as_deref().expose()).with_context(|| {
+            format!(
+                "Token for {sid} is invalid (defined in {})",
+                token.definition
+            )
+        })?;
+    }
```

Comment:
This is moving the check from
- token credential provider

To
- token credential provider
- paseto credential provider
- the enumeration of all credential providers

I'd want to better understand if its a good idea to be running this check in all of those other situations

CC @arlosi

Reply:
Construction of `RegistryConfig` is closer to TOML deserialization of `credentials.toml` than to implementation of credential providers. The token here is the *input* for downstream users of the registry config, and does not touch *output* token of credential providers (that's `CredentialResponse`).

Paseto uses `secret_key` and `secret_key_subject` fields, not `token`.

This field is only read by `TokenCredential`, apart from code that warns when `cargo::token` is not enabled.

Reply:
@arlo I'd be interested in your thoughts on this